ion-hash/src/lib.rs

```diff
@@ -0,0 +1,120 @@
+// Copyright Amazon.com, Inc. or its affiliates.
+
+//! Implement Ion Hash as per the [spec](https://amzn.github.io/ion-hash/docs/spec.html)
```

Comment:
Minor doc comment edit:

```suggestion
//! Implements the [Ion Hash specification](https://amzn.github.io/ion-hash/docs/spec.html)
```